### PR TITLE
[3006.x] Preserve test condition during sub-state runs

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Preserve `__opts__["test"]` value for sub states.

### What issues does this PR fix or reference?
Fixes: #62590 #64195 (alternate to #64269)

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### CODE REVIEW

File: tests/pytests/integration/states/test_x509_v2.py [hunk 1]
```diff
@@ -46,6 +46,65 @@ def x509_pkidir(tmp_path_factory):
...
+    Another private key is (already) present:
+      x509.private_key_managed:
+        - name: {existing_privkey}
+    Subsequent file.managed call should not run in test mode:
+      file.managed:
+        - name: {new_file}
+        - contents: foo
+        - require:
+          - Another private key is (already) present
+    """
```
Comment (s0undt3ch) [inline tests/pytests/integration/states/test_x509_v2.py:91]:
```suggestion
    state = f"""
    # The result of this call is irrelevant, just that it exists
    Some private key is present:
      x509.private_key_managed:
        - name: {new_privkey}
    # This single call without changes does not trigger the bug on its own
    Another private key is (already) present:
      x509.private_key_managed:
        - name: {existing_privkey}
    Subsequent file.managed call should not run in test mode:
      file.managed:
        - name: {new_file}
        - contents: foo
        - require:
          - Another private key is (already) present
    """
```

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/integration/states/test_x509_v2.py (hunk 1 above)
Comment (dwoz) [reply tests/pytests/integration/states/test_x509_v2.py]: Thanks, I copied this test from #64269. I'm not sure what, if anything, removing the `\` changes. Other than the fact that it's not needed.

Comment (s0undt3ch) [reply tests/pytests/integration/states/test_x509_v2.py]: It's just not really needed. That's all.

**Review by Ch3LL — DISMISSED** [review]
Code looks good, although there are some failing tests

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**